includes/class-edd-discount.php

```diff
@@ -0,0 +1,583 @@
+		 */
+		return apply_filters( 'edd_get_discount_expiration', $this->end_date, $this->ID );
+	}
+
+	/**
+	 * Retrieve the maximum uses for the discount code.
+	 *
+	 * @since 2.7
+	 * @access public
+	 *
+	 * @return int Maximum uses.
+	 */
+	public function get_max_uses() {
+		$this->max_uses = get_post_meta( $this->ID, '_edd_discount_max_uses', true );
+
+		/**
+		 * Filters the maximum uses.
+		 *
+		 * @since 2.7
+		 *
+		 * @param int $max_uses Maximum uses.
+		 * @param int $ID       Discount ID.
+		 */
+		return (int) apply_filters( 'edd_get_discount_max_uses', $this->max_uses, $this->ID );
+	}
+
+	/**
+	 * Retrieve the minimum spend required for the discount to be satisfied.
+	 *
+	 * @since 2.7
+	 * @access public
+	 *
+	 * @return mixed float Minimum spend.
+	 */
+	public function get_min_amount() {
+		$this->min_amount = get_post_meta( $this->ID, '_edd_discount_min_price', true );
+
+		/**
+		 * Filters the minimum amount.
+		 *
+		 * @since 2.7
+		 *
+		 * @param float $min_amount Minimum amount.
+		 * @param int   $ID         Discount ID.
+		 */
+		return (float) apply_filters( 'edd_get_discount_min_price', $this->min_amount, $this->ID );
+	}
+
+	/**
+	 * Retrieve the usage limit per limit (if the discount can only be used once per customer).
+	 *
+	 * @since 2.7
+	 * @access public
+	 *
+	 * @return bool Once use per customer?
+	 */
+	public function get_is_single_use() {
+		$this->is_single_use = get_post_meta( $this->ID, '_edd_discount_is_single_use', true );
+
+		/**
+		 * Filters the single use meta value.
+		 *
+		 * @since 2.7
+		 *
+		 * @param bool $is_single_use Is the discount only allowed to be used once per customer.
+		 * @param int  $ID            Discount ID.
+		 */
+		return (bool) apply_filters( 'edd_is_discount_single_use', $this->is_single_use, $this->ID );
+	}
+
+	/**
+	 * Retrieve the property determining if a discount is not global.
+	 *
+	 * @since 2.7
+	 * @access public
+	 *
+	 * @return bool Whether or not the discount code is global.
+	 */
+	public function get_is_not_global() {
+		$this->is_not_global = get_post_meta( $this->ID, '_edd_discount_is_not_global', true );
+
+		/**
+		 * Filters if the discount is global or not.
+		 *
+		 * @since 2.7
+		 *
+		 * @param bool $is_not_global Is the discount global or not.
+		 * @param int  $ID            Discount ID.
+		 */
+		return (bool) apply_filters( 'edd_discount_is_not_global', $this->is_not_global, $this->ID );
+	}
+
+	/**
+	 * Helper function to get discounts by a meta key and value provided.
+	 *
+	 * @since 2.7
+	 * @access public
+	 *
+	 * @param string $key   Value of the meta key to retrieve.
+	 * @param string $value Meta value for the key passed.
+	 * @return mixed array|bool
+	 */
+	public function get_by( $field = '', $value = '' ) {
+		if ( empty( $field ) || empty( $value ) ) {
+			return false;
+		}
+
+		if ( ! is_string( $field ) ) {
+			return false;
+		}
+
+		switch ( strtolower( $field ) ) {
+			case 'code':
+				break;
+
+			case 'id':
```

Comment:
We should be really careful about this. Do we really need to allow post ID querying? If we're moving to a custom table this could get really confusing if its just called "id". Imagine a year from now if these are stored in custom tables. Is this a row ID in the db table or the post ID, despite the fact we won't be storing them in posts anymore.